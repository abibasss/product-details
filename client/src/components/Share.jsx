import React from 'react';

const Share = (props) => {
  return (
    <div>
      <div className="row">
        <h1> SHARE HOW YOU WEAR IT </h1>
        <p className="share">Share a photo of your adidas favorite and appear in our showcase
        below. Make sure to tag your image with @adidas </p>
      </div>
      <div className="col-2-12justify-content-center share-photo">
        <ul>
          <li><div><img className= "lrg-share img-fluid overlay-blk" src="https://photorankmedia-a.akamaihd.net/media/8/y/w/8ywuwc4/mobile.jpg"></img></div></li>
        </ul>
        <ul>
          <li><div> <img className="sm-share img-fluid" src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"></img></div></li>
          <li><div> <img className="sm-share img-fluid" src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"></img></div></li>
        </ul>
        <ul>
          <li><div> <img className="sm-share img-fluid" src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"></img></div></li>
          <li><div> <img className="sm-share img-fluid" src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"></img></div></li>
        </ul>
      </div>
    </div>
  )
}


export default Share;
