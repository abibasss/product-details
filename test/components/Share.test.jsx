import React from 'react';
import { shallow } from 'enzyme';
import Share from '../../client/src/components/Share';

let sample = {
  "id": 1,
  "user1": "anonymouse",
  "img1": "https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg",
  "user2": "indigosparkle",
  "img2": "https://s3-us-west-1.amazonaws.com/adidas-shoe/share7.jpg",
  "user3": "weareliving",
  "img3": "https://s3-us-west-1.amazonaws.com/adidas-shoe/share1.jpg",
  "user4": "urk3wl",
  "img4": "https://s3-us-west-1.amazonaws.com/adidas-shoe/share5.jpg",
  "user5": "cereal_killer",
  "img5": "https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg"
}

describe('Share Component', () => {
  let wrapper = shallow(<Share shares={sample} />);

  it('should look like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 5 different pictures/users', () => {
    expect(wrapper.find('div.overlay-blk').length).toBe(5);
  });

  it('should not show username on normal display', () => {
    expect(wrapper.find('span').at(0)).toHaveClassName('hidden');
    expect(wrapper.find('span').at(3)).toHaveClassName('hidden');
  });

  it('should display username on hover/mouseenter', () => {
    wrapper.find('div.overlay-blk').at(0).prop('onMouseEnter')();
    expect(wrapper.find('span').at(0).hasClass('hidden')).toBe(false);
  });

  it('should display correct username on hover', () => {
    wrapper.find('div.overlay-blk').at(4).prop('onMouseEnter')();
    expect(wrapper.find('span').at(4).hasClass('hidden')).toBe(false);
    expect(wrapper.find('span').at(4).text()).toBe('cereal_killer');
  })

  it('should not display username on mouse leave', () => {
    wrapper.find('div.overlay-blk').at(1).prop('onMouseEnter')();
    expect(wrapper.find('span').at(1).hasClass('hidden')).toBe(false);

    wrapper.find('div.overlay-blk').at(1).prop('onMouseLeave')();
    expect(wrapper.find('span').at(1)).toHaveClassName('hidden');
  });
})
