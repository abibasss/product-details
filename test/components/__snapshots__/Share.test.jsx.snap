// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Component should look like snapshot 1`] = `
<div
  className="share"
>
  <div
    className="title-1000"
  >
    <h1>
       SHARE HOW YOU WEAR IT 
    </h1>
    <p
      className="share-desc"
    >
      Share a photo of your adidas favorite and appear in our showcase below. Make sure to tag your image with @adidas
    </p>
  </div>
  <div
    className="col-2-12 justify-content-center share-photo"
  >
    <ul>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <img
            className="lrg-share img-fluid"
            src="https://photorankmedia-a.akamaihd.net/media/8/y/w/8ywuwc4/mobile.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          anonymouse
        </span>
      </li>
    </ul>
    <ul>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <img
            className="sm-share img-fluid"
            src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          indigosparkle
        </span>
      </li>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
           
          <img
            className="sm-share img-fluid"
            src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          weareliving
        </span>
      </li>
    </ul>
    <ul>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
           
          <img
            className="sm-share img-fluid"
            src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          urk3wl
        </span>
      </li>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
           
          <img
            className="sm-share img-fluid"
            src="https://z3photorankmedia-a.akamaihd.net/media/3/f/7/3f7ywc4/mobile.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          cereal_killer
        </span>
      </li>
    </ul>
  </div>
</div>
`;
